iap/request.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package iap contains Identity-Aware Proxy samples.
+package iap
+
+// [START iap_make_request]
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"google.golang.org/api/idtoken"
+)
+
+// makeIAPRequest makes a request to an application protected by Identity-Aware
+// Proxy with the given iapClientID.
+func makeIAPRequest(w io.Writer, request *http.Request, iapClientID string) error {
+	// request, err := http.NewRequest("GET", "http://example.com", nil)
+	// iapClientID := "IAP_CLIENT_ID.apps.googleusercontent.com"
```

Comment:
Should we name this variable something else? Seems like this is a URL that uses the client ID.

Reply:
I was going for consistency with other samples, but I think you are right. Will switch to audience as that is what it actually is.